### PR TITLE
Prox [WIP]

### PR DESCRIPTION
**Pull request by ingalls**
Add proximity support to mapbox.js

### CODE REVIEW

File: src/geocoder.js [hunk 1]
```diff
@@ -35,12 +40,16 @@ module.exports = function(url, options) {
...
-        return L.Util.template(geocoder.getURL(), {query: query});
+
+        return L.Util.template(geocoder.getURL(), {
+            proximity: args.proximity ? encodeURIComponent(args.proximity[0]+','+args.proximity[1]) : false,
+            query: query 
+        });
     };
 
-    geocoder.query = function(_, callback) {
+    geocoder.query = function(_, args, callback) {
```
Comment (jfirebaugh) [inline src/geocoder.js]: [`query` is a public API](https://www.mapbox.com/mapbox.js/api/v2.1.9/l-mapbox-geocoder/#section-geocoder-query) -- we can't make a backwards incompatible change to the method signature until 3.0.

We could make the first parameter accept an object as well as a plain string, and support:

`geocoder.query({query: 'austin', proximity: latlng}, callback)`

WDYT?

File: src/geocoder_control.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ var GeocoderControl = L.Control.extend({
     includes: L.Mixin.Events,
 
     options: {
+        proximity: false,
```
Comment (jfirebaugh) [inline src/geocoder_control.js]: Let's default this to `true`.

File: src/geocoder.js (hunk 1 above)
Comment (ingalls) [inline src/geocoder.js]: :+1:

File: src/geocoder_control.js (hunk 2 above)
Comment (ingalls) [inline src/geocoder_control.js]: To me that would be a breaking change to the API as one is going to get drastically different results than with it set to false. Would it be better to save this for `3.0.0`?

Comment (jfirebaugh) [inline src/geocoder_control.js]: Hopefully drastically better results though, right? But, yeah, I can see an argument either way, and it's easy to switch later on. @tmcw want to be a tiebreaker?

Comment (tmcw) [inline src/geocoder_control.js]: I don't think this justifies a 3.0.0: it's not a breaking change, it's an improvement. This fits as a minor version bump.

Comment (ingalls) [inline src/geocoder_control.js]: :+1:

**Pull request closed without merging**